### ISSUE TITLE
clang-tidy bugprone-argument-comment check template arguments

### ISSUE THREAD

**Issue by MatzeB** [llvm/llvm-project#173045]
Catching the problem in the following example would be a useful addition to the `bugprone-argument-comment` rule:

```
template<typename Input, typename Output> class C {};
int main() {
  // Should produce a warning, as order is swapped:
  C</*Output=*/float, /*Input=*/int> xxx;
}
```

currently produces no warning:

```
$ clang-tidy --checks=bugprone-argument-comment ./test.cpp
... (no warning printed)
```
